### PR TITLE
Add the CI for auto release the Helm chart

### LINKED ISSUE #2278: [CI] Incorporate creation and publishing of Meshery Helm chart into stable release workflow
#### Current Behavior
- Location of source code for Meshery’s chart: https://github.com/layer5io/meshery/tree/master/install/kubernetes/helm/meshery
- Location of published Meshery chart: https://github.com/layer5io/meshery.io/tree/master/charts ([used by Artifact Hub](https://artifacthub.io/packages/helm/meshery/meshery))
- No workflow included in the build and release CI workflow for publishing the helm chart.

#### Desired Behavior
1. The Helm chart should be incorporated into the `stable` release channel.
1. The Helm chart should follow Meshery’s versioning.

#### Implementation
- No workflow included in the build and release CI workflow for publishing the helm chart.

#### Acceptance Tests
1. Meshery's helm chart is updated and published with each Meshery stable version release.
1. Meshery's helm chart no longer has its own version number, but follows that of Meshery (server).

**_Future:_**
1. Users can use `mesheryctl system channel` to switch between release channels. // @anirudhjain75 @navendu-pottekkat

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: aisuko <urakiny@gmail.com>

**Description**

This PR fixes #2278 

**Notes for Reviewers**

The CI detect the file change in the folder under `install/kubernetes/helm` path. Every time the file change the CI will run two processor
* Check the file changed if not
* Lint the Helm charts syntax
* If pass the Lint, the CI will upload the new version chart package to the `layer5io/meshery.io` release page and push the new charts index.yaml to the `master` branch `charts` folder.

Requirement:

* We need to have a token of the Personal development scope of layer5io and name `secrets.GITHUB_TOKEN`
* The action commit user email can be change

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by kumarabd** [comment]
@Aisuko Two things to note here:
1. We would need to account for edge releases, so a separate artifact repository for edge releases would be a preferred option here.
2. To version the helm charts in sync with the meshery version, so that we dont need to maintain separate versions for the charts and also users dont get confused.
Do these above feedback makes sense to you?

**Comment by Aisuko** [comment]
> @Aisuko Two things to note here:
> 
> 1. We would need to account for edge releases, so a separate artifact repository for edge releases would be a preferred option here.
> 2. To version the helm charts in sync with the meshery version, so that we dont need to maintain separate versions for the charts and also users dont get confused.
>    Do these above feedback makes sense to you?
> 
> Examples:
> https://artifacthub.io/packages/helm/linkerd2/linkerd-jaeger for stable
> https://artifacthub.io/packages/helm/linkerd2-edge/linkerd-jaeger for edge releases

Thanks, @kumarabd , that makes a lot of sense to me. I will do PR for this.

**Review by ramrodo — APPROVED** [review]
Great! LGTM

File: .github/workflows/helm-ci.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          path: ${{ env.REPO_PATH }}
+      - name: Copy index to new repo
+        env:
+          SOURCE: /home/runner/work/meshery/meshery/.cr-index/index.yaml
+          TARGET: meshery.io/charts
+        run:
+          cp ${{ env.SOURCE }} ${{ env.TARGET }}
+      - name: Push
+        env:
+          REPO_PATH: meshery.io
```
Comment (leecalcote) [inline .github/workflows/helm-ci.yml:90]: I wonder if this will have to be `layer5io/meshery.io`

**Review by leecalcote — APPROVED** [review]

**Pull request merged**